pkg/edgeview/src/system.go

```diff
@@ -0,0 +1,766 @@
+				data.HashAlgo, data.Type, base64.StdEncoding.EncodeToString(data.CertHash))
+			printCert(data.Cert)
+		}
+	}
+}
+
+func printCert(certdata []byte) {
+	block, _ := pem.Decode(certdata)
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		fmt.Printf("cert err %v\n", err)
+		return
+	}
+	fmt.Printf("    subject: %s, serial: %d, valid until: %v\n", cert.Subject, cert.SerialNumber, cert.NotAfter)
+	fmt.Printf("     issuer: %s\n", cert.Issuer)
+}
+
+func runConfigItems() {
+	printColor(" - global settings:", colorCYAN)
+
+	// /hostfs is need here to read the memory limit for eve item
+	configitems := getConfigItems()
+
+	configMap := types.NewConfigItemSpecMap()
+	// global
+	for k, g := range configitems.GlobalSettings {
+		//fmt.Printf("key: %s; value %v\n", k, g)
+		s := configMap.GlobalSettings[k]
+		s1 := types.ConfigItemValue{
+			ItemType: s.ItemType,
+			IntValue: s.IntDefault,
+			StrValue: s.StringDefault,
+			BoolValue: s.BoolDefault,
+			TriStateValue: s.TriStateDefault,
+		}
+		if getCfgValue(g) == getCfgValue(s1) {
+			buff := fmt.Sprintf(" %s: %s\n", k, getCfgValue(g))
+			fmt.Printf("%s", buff)
+		} else {
+			buff := fmt.Sprintf(" %s: %s; default %s\n", k, getCfgValue(g), getCfgValue(s1))
+			printColor(buff, colorYELLOW)
+		}
+		//fmt.Printf("   default: %v\n", configMap.GlobalSettings[k])
+	}
+
+	// agent
+	printColor("\n - agent settings:", colorCYAN)
+	for k, g := range configitems.AgentSettings {
+		printColor("  "+k+":  ", colorRED)
+		for k1, g1 := range g {
+			fmt.Printf("    %s, %s\n", k1, getCfgValue(g1))
+		}
+	}
+}
+
+func getConfigItems() types.ConfigItemValueMap {
+	var cfgItem types.ConfigItemValueMap
+	retStr, err := runCmd("cat /persist/status/zedagent/ConfigItemValueMap/global.json", false, false)
+	if err != nil {
+		return cfgItem
+	}
+	_ = json.Unmarshal([]byte(retStr), &cfgItem)
+	return cfgItem
+}
+
+func getCfgValue(g types.ConfigItemValue) string {
+	value := ""
+	switch g.ItemType {
+	case types.ConfigItemTypeInt:
+		value = strconv.Itoa(int(g.IntValue))
+	case types.ConfigItemTypeBool:
+		value = strconv.FormatBool(g.BoolValue)
+	case types.ConfigItemTypeString:
+		value = g.StrValue
+	case types.ConfigItemTypeTriState:
+		if g.TriStateValue == types.TS_NONE {
+			value = "None"
+		} else if g.TriStateValue == types.TS_DISABLED {
+			value = "Disabled"
+		} else if g.TriStateValue == types.TS_ENABLED {
+			value = "Enabled"
+		} else {
+			value = "un-supported"
+		}
+	default:
+		value = "un-supported"
+	}
+	return value
+}
+
+func getDownload() {
+	pubsubSvs("/run/", "volumemgr", "DownloaderConfig")
+	pubsubSvs("/run/", "downloader", "DownloaderStatus")
+
+	getMetricsMap("/run/downloader/MetricsMap/", nil, true)
+	checkDownload("/persist/downloads")
+	checkDownload("/persist/vault/downloader")
+	checkDownload("/persist/vault/verifier")
+}
+
+func runPS(opt string) {
+	var item string
+	if strings.Contains(opt, "ps/") {
+		opts := strings.SplitN(opt, "ps/", 2)
+		item = opts[1]
```

Comment:
Is this to look for a particular command string? If so can we validate that it is an alphanumeric string (a-z0-9 allowed)?

Reply:
will do.